test/testsuites/migrationtests.go

```diff
@@ -0,0 +1,68 @@
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// MigrationTests is a test suite that ensures the following:
+// * All stale datastore CatalogSourceConfigs created by marketplace after the creation of
+// OperatorSources in a 4.1.x cluster are deleted when the cluster is migrated to openshift 4.2.0.
+// * Installed CatalogSourceConfigs created during operator installation in a 4.1 cluster are deleted.
+// * Existing Subscriptions are updated to reference the datastore CatalogSources instead
+// of installed CatalogSources created during operator installation in a openshift 4.1.x cluster.
+func MigrationTests(t *testing.T) {
+	t.Run("catalogsourceconfigs-are-cleaned-up", testCatalogSourceConfigsCleanedUp)
```

Comment:
We should test the negative as well - make sure that a Subscription that doesn't point to a CSC's CatalogSource is not touched in any way.